sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -186,6 +198,7 @@ private[sql] class JSONOptions(
         JsonReadFeature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER,
         allowBackslashEscapingAnyCharacter)
       .configure(JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS, allowUnquotedControlChars)
+      .streamReadConstraints(new StreamReadConstraints(maxNestingDepth, maxNumLen, maxStringLen))
```

Comment:
could you use StreamReadConstraints.builder and use the fluent style of setting the configs? - much more explicit and readable